src/main/java/com/amazon/opendistroforelasticsearch/sql/rewriter/nestedfield/NestedFieldProjection.java

```diff
@@ -57,14 +62,21 @@ public NestedFieldProjection(SearchRequestBuilder request) {
     public void project(List<Field> fields) {
         if (isAnyNestedField(fields)) {
             initBoolQueryFilterIfNull();
-
+            // Extract nested query from WHERE clause
             List<NestedQueryBuilder> nestedQueries = extractNestedQueries(query());
             groupFieldNamesByPath(fields).forEach(
-                (path, fieldNames) -> buildInnerHit(fieldNames, findNestedQueryWithSamePath(nestedQueries, path))
+                (path, fieldNames) -> buildInnerHit(fieldNames, findNestedQueryWithSamePath(nestedQueries, path, fieldNames))
             );
         }
     }
 
+    /** Reset values of lastFieldInPath and lastFieldNames to prepare for next check */
+    private void setLastValues(String path, List<String> fieldNames) {
```

Comment:
rename to avoid unclear `values`? maybe `setLastField...`?

Reply:
will change to `setLastField`